src/lib/y2network/hostname_reader.rb

```diff
@@ -94,5 +122,26 @@ def random_hostname
       suffix = HOSTNAME_CHARS.sample(4).join
       "linux-#{suffix}"
     end
+
+  private
+
+    # Runs workflow for querying hostname in the installer
+    #
+    # @return [String] Hostname
+    def hostname_for_installer
+      @install_inf_hostname = hostname_from_install_inf
```

Comment:
why only this part is catched and others not?

Reply:
see bellow comment about separating hostname class, this is another reason. In fact Hostname and DNS has nothing in common. It stayed together just because it was together in the old code.